distogram/__init__.py

```diff
@@ -373,57 +335,62 @@ def histogram(h, ucount=100):
     Raises:
         ValueError if distribution contains less elements than the number of
         bins in the Distogram object.
-    '''
+    """
     if len(h.bins) < h.bin_count:
         raise ValueError("not enough elements in distribution")
 
-    last = 0.0
-    u = []
-    bounds = _linspace(h.min, h.max, num=ucount+1)
-    for e in bounds[1:-1]:
-        new = count_at(h, e)
-        u.append((e, new-last))
-        last = new
+    bin_bounds = _linspace(h.min, h.max, num=ucount+1)
+    counts = [count_at(h, e) for e in bin_bounds[1:-1]]
+    u = [(bin_bounds[0], counts[0])]
+    u.extend([
+        (b, new - last)
+        for b, new, last in zip(bin_bounds[1:], counts[1:], counts[:-1])
+    ])
+
     return u
 
 
-def quantile(h, value):
-    '''Returns a quantile of the distribution
+def quantile(h: Distogram, value: float) -> Optional[float]:
+    """ Returns a quantile of the distribution
 
     Args:
         h: A Distogram object.
         value: The quantile to compute. Must be between 0 and 1
 
     Returns:
         An estimation of the quantile. Returns None if the Distogram
-        object contains no element or value is outside of [0:1].
-    '''
+        object contains no element or value is outside of (0:1).
+    """
     if len(h.bins) == 0:
         return None
 
-    if value < 0.0 or value > 1.0:
+    if not (0.0 < value < 1.0):
```

Comment:
Values 0.0 and 1.0 are valid. It should be:
```
if not (0.0 <= value <= 1.0):
```

Reply:
fixed